iocore/net/SSLUtils.cc

```diff
@@ -1599,6 +1672,15 @@ SSLMultiCertConfigLoader::_set_alpn_callback(SSL_CTX *ctx)
   return true;
 }
 
+bool
+SSLMultiCertConfigLoader::_set_keylog_callback(SSL_CTX *ctx)
+{
+#if OPENSSL_VERSION_NUMBER >= 0x010100000
```

Comment:
`OPENSSL_VERSION_NUMBER` is deprecated since OpenSSL 3.0. Let's not use it for new code.

Reply:
I don't mind changing to something else, but don't we need to use `OPENSSL_VERSION_NUMBER` as long as we support OpenSSL 1.1.x? Users building with OpenSSL 1.1.1 won't have access to the newer macros/functions.

Maybe I'm missing something.

Reply:
As we talked on Slack, I'd check availability of the function in configure script. That way should always work for any versions and also BoringSSL.

Reply:
👍 

Good suggestion. I added the check to the configure mechanism as suggested.